app/product/[title]/page.tsx

```diff
@@ -0,0 +1,153 @@
+"use client";
+import React, { useEffect, useState } from "react";
+import { notFound, useSearchParams } from "next/navigation";
+import { useAppDispatch, useAppSelector } from "@/lib/hookts";
+import { RootState } from "@/lib/store";
+import { fetchProducts } from "@/lib/features/products/FetchDataSlice";
+import { Product } from "@/lib/types/types";
+import ReferenceComments from "@/components/infoModal/ReferenceComments";
+import AddProductButton from "@/components/buttons/AddProduct";
+import Image from "next/image";
+import { FaDotCircle } from "react-icons/fa";
+import { BsFillHouseDownFill } from "react-icons/bs";
+import { TbMathGreater } from "react-icons/tb";
+import Link from "next/link";
+import RatingComponent from "@/utils/ReactStars";
+import ClipLoader from "react-spinners/ClipLoader";
+
+export default function ProductPage({ params }: { params: { title: string } }) {
+  const dispatch = useAppDispatch();
+  const products = useAppSelector(
+    (state: RootState) => state.products.products
+  );
+  const searchParams = useSearchParams();
+  const productId = searchParams.get("productId");
+  const [product, setProduct] = useState<Product | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  useEffect(() => {
```

Comment:
@Kamildeeal don't define any functions inside `useEffect`. Also move all of this logic to a separate hook